File: ehs_backend/app/schemas/hazard.py
from pydantic import BaseModel


class ControlMeasure(BaseModel):
    hierarchy: str        # "Elimination" | "Substitution" | "Engineering" | "Administrative" | "PPE"
    measures: list[str]   # each item is one bullet with inline WSH ref


class MitigationActivity(BaseModel):
    activity: str
    responsible_party: str
    priority: str         # "Immediate" | "Short-term" | "Ongoing"


class HazardAnalysisResponse(BaseModel):
    hazard_identified: str
    hazard_description: str
    risk_level: str                        # e.g. "High (L4 × S4 = 16)"
    potential_consequences: list[str]
    control_measures: list[ControlMeasure]
    mitigation_activities: list[MitigationActivity]
    applicable_regulations: list[str]
    residual_risk: str                     # e.g. "Medium (L2 × S4 = 8)"

File: ehs_frontend/src/api/hazard.ts
import { api } from './client'

export interface ControlMeasure {
  hierarchy: string
  measures: string[]
}

export interface MitigationActivity {
  activity: string
  responsible_party: string
  priority: 'Immediate' | 'Short-term' | 'Ongoing'
}

export interface HazardAnalysisResponse {
  hazard_identified: string
  hazard_description: string
  risk_level: string
  potential_consequences: string[]
  control_measures: ControlMeasure[]
  mitigation_activities: MitigationActivity[]
  applicable_regulations: string[]
  residual_risk: string
}

export async function analyseHazard(text: string, image?: File): Promise<HazardAnalysisResponse> {
  const form = new FormData()
  if (text) form.append('text', text)
  if (image) form.append('image', image)
  const { data } = await api.post('/api/v1/hazard/analyse', form, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
  return data
}
